test/modules/MockModule.ts

```diff
@@ -43,3 +50,64 @@ export function registerMockModule(): MockModule {
     }
     return module;
 }
+
+export class MockModuleWithCryptoSetupExtension extends RuntimeModule {
+    public get apiInstance(): ModuleApi {
+        return this.moduleApi;
+    }
+
+    moduleName: string = MockModuleWithCryptoSetupExtension.name;
+
+    extensions: AllExtensions = {
+        cryptoSetup: new (class extends CryptoSetupExtensionsBase {
+            SHOW_ENCRYPTION_SETUP_UI = true;
+
+            examineLoginResponse(response: any, credentials: ExtendedMatrixClientCreds): void {
+                throw new Error("Method not implemented.");
+            }
+            persistCredentials(credentials: ExtendedMatrixClientCreds): void {
+                throw new Error("Method not implemented.");
+            }
+            getSecretStorageKey(): Uint8Array | null {
+                return Uint8Array.from([0x11, 0x22, 0x99]);
+            }
+            createSecretStorageKey(): Uint8Array | null {
+                throw new Error("Method not implemented.");
+            }
+            catchAccessSecretStorageError(e: Error): void {
+                throw new Error("Method not implemented.");
+            }
+            setupEncryptionNeeded(args: CryptoSetupArgs): boolean {
+                throw new Error("Method not implemented.");
+            }
+            getDehydrationKeyCallback():
+                | ((
+                      keyInfo: SecretStorageKeyDescriptionAesV1,
+                      checkFunc: (key: Uint8Array) => void,
+                  ) => Promise<Uint8Array>)
+                | null {
+                throw new Error("Method not implemented.");
+            }
+        })(),
```

Comment:
If you don't plan to call any of these methods, it might be simpler just to type-cast an empty object:

```suggestion
        cryptoSetup: {} as unknown as CryptoSetupExtensionsBase,
```

(If you *do* plan to call them, I suggest using `jest.fn()` to create mock methods, so that tests can change the behaviour and check if they were called.)

Reply:
Good point. I have simplified using using jest.fn()